src/core_ocean/analysis_members/mpas_ocn_moc_streamfunction.F

```diff
@@ -125,7 +125,7 @@ subroutine ocn_init_moc_streamfunction(domain, err)!{{{
       integer :: currentRegion, i, iCell
 
       !! region arrays/variables
-      character (len=STRKIND), dimension(:), pointer :: regionGroupNames
+      character (len=ShortStrKIND), dimension(:), pointer :: regionGroupNames
```

Comment:
@vanroekel why is this changed to `ShortStrKIND`?  It makes compile fail at
```
mpas_ocn_moc_streamfunction.F:150:80:
       call mpas_pool_get_array(regionPool, 'regionGroupNames', regionGroupNames)
```
because this subroutine only supports STRKIND. When I change it back it compiles, but I wonder if you changed it for another reason.

Reply:
I thought this was needed as the regionGroupName variable in the MOC transect file was ShortStrKINd (64) and the routine wasn't reading that field.  Can you try a test with the AM on and reverted?  

Related, what do you mean it only supports STRKND?  I was able to run this in e3sm with that setting with no issue.  Does it only come up in debug or something?

Reply:
This PR compiles and runs with intel 18 (debug and opt) but produces that compile error with gnu 5.3 on grizzly (debug and opt).

I don't see anywhere that shortStrKind is used in the ocean core. On `ocean/develop`:
```
cd src/core_ocean
git grep -in shortstrkind
```
produces nothing, and 
```
git grep -in regionGroupName|grep character
analysis_members/mpas_ocn_meridional_heat_transport.F:132:      character (len=STRKIND), dimension(:), pointer :: regionGroupNames
analysis_members/mpas_ocn_meridional_heat_transport.F:344:      character (len=STRKIND), dimension(:), pointer :: regionNames, regionGroupNames
analysis_members/mpas_ocn_moc_streamfunction.F:128:      character (len=STRKIND), dimension(:), pointer :: regionGroupNames
analysis_members/mpas_ocn_moc_streamfunction.F:358:      character (len=STRKIND), dimension(:), pointer :: regionNames, regionGroupNames
analysis_members/mpas_ocn_water_mass_census.F:208:      character (len=STRKIND), dimension(:), pointer :: regionNames, regionGroupNames
```
shows everything is `STRKIND`. I'll go ahead and change that one back. 

BTW, I compiled E3SM on cori, and both gnu and intel compiled fine, so it must be version dependent. 

What I meant above is that the general interface for `mpas_pool_get_array` only supports two character versions:
```
cd src/framework
git grep -in mpas_pool_get_array|grep subrou|grep '{{{'|grep char
mpas_pool_routines.F:4480:   subroutine mpas_pool_get_array_0d_char(inPool, key, string, timeLevel)!{{{
mpas_pool_routines.F:4510:   subroutine mpas_pool_get_array_1d_char(inPool, key, array, timeLevel)!{{{
```
and in both of those, the string is defined as a `strKind`:
```
4486       character (len=StrKIND), pointer :: string
4516       character (len=StrKIND), dimension(:), pointer :: array
```
So gnu 5.3 is correct to complain about it.